raster-foundry/raster-foundry
# Enable lab layer selection

## Overview

This PR:
- enables layer selection in lab
- updates lab node display to include nodes with type `layerSrc`
- reuses project select modal to allow users to select a layer after selecting a project
- updates project item component to map project layers as well
- updates project service to include getting layer scene statuses

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

A fresh analysis:
<img width="1521" alt="screen shot 2019-03-05 at 4 49 34 pm" src="https://user-images.githubusercontent.com/16109558/53840489-4c413000-3f68-11e9-995c-2125a181d06a.png">

Select project layer based on project selection:
<img width="1521" alt="screen shot 2019-03-05 at 4 50 49 pm" src="https://user-images.githubusercontent.com/16109558/53840515-595e1f00-3f68-11e9-9e16-5cfce4c28f74.png">

View results generated from picked projects, layers, and bands:
<img width="1521" alt="screen shot 2019-03-05 at 4 51 34 pm" src="https://user-images.githubusercontent.com/16109558/53840556-71ce3980-3f68-11e9-869e-bf70f9605f71.png">


## Testing Instructions

 * Create a project with multiple layers and scenes in them
 * Go to an old analysis, make sure it uses project default layer, make sure the nodes and maps display work
 * Create a new analysis with the project created in step 1, make sure default layer is selected after selecting a project, change layer selection, update bands, visualize input nodes, visualize calculated nodes, update layer selection, etc.

Closes https://github.com/raster-foundry/raster-foundry/issues/4605
